Unpack Kubernetes config

pidster opened the pull request:
Fixes: https://github.com/microservices-demo/microservices-demo/issues/310

amouat committed:
Fix typo

Fix a very minor typo (Sho -> Shop)

idcrosby committed:
fix marathon file

idcrosby committed:
remove login

idcrosby committed:
use latest tag

pidster committed:
Initial split

errordeveloper commented on deploy/kubernetes/manifests/v1/ksNamespace.yaml:
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kube-system
```
I kept wondering why do we have to define this?

errordeveloper replied:
I think it's an artefact of something that we didn't have to do the first place, also the way namespaces are created has changed recently also. It's okay keep it, if we have scripts that refer to it, and remove when we move to `kubeadm`...

errordeveloper commented on deploy/kubernetes/manifests/complete-demo.yaml:
```diff
@@ -0,0 +1,437 @@
+    name: user
+spec:
+  ports:
+    # the port that this service should serve on
```
These comments seem quite redundant, especially because we aren't doing anything worse explaining here, and it's not like we comment on other fields (if we were to attempt to explain Kubernetes).

errordeveloper commented on deploy/kubernetes/manifests/weavescope.yaml:
```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
```
Why do we need this file here? The docs reference launcher URL, which is what we should do.

errordeveloper reviewed (commented):
Overall this is much better then how we had all of these in one directory. I am not sure about having `v1` and `v1beta1` is so appropriate. It seems as if versioned directories refer to application versions, and they aren't... Could you move those into one subdirectory and simply call it `app`, with `compete-demo.yaml` renamed to `app-combined.yaml` to match?

pidster committed:
Correct a misunderstanding about splitting parts.

Removed a split-by-api-version and mark Deployments and Services separately, all in the same directory.

pidster committed:
Use a better nomenclature.

squaremo commented on deploy/kubernetes/manifests/catalogue-db-dep.yaml:
```diff
@@ -0,0 +1,19 @@
+apiVersion: extensions/v1beta1
```
In general it's useful to put `---` at the top of yamels, so they can be 'cated together.

pidster replied:
Noted, will add this.

errordeveloper reviewed (approved):
LGTM, let's discuss whether we need `weavescope.yaml` separately.

squaremo commented on deploy/kubernetes/weavescope.yaml:
```diff
@@ -0,0 +1,79 @@
+apiVersion: v1
```
Is this supposed to be here?

I don't think it'll work with fluxy, which expects either Deployment or ReplicationController resources. But it might not break it.

pidster replied:
Ack. Not applicable to flux.

squaremo commented on deploy/mesos-marathon/marathon.json:
```diff
@@ -5,18 +5,18 @@
       "id": "/weave-demo/database",
```
Does this change definitely belong in this PR? It seems unrelated.

pidster replied:
Rebase pulled it in.

squaremo commented on deploy/kubernetes/complete-demo.yaml:
```diff
@@ -0,0 +1,437 @@
+apiVersion: extensions/v1beta1
```
I don't think you need to keep the all-in-one file around.

pidster replied:
The all-in-one is separate use-case, as @errordeveloper says.

pidster committed:
Merge branch 'master' into unpack_k8s_yaml

pidster committed:
Add handy file separator that aids concatenation

pidster committed:
Merge branch 'master' into unpack_k8s_yaml
